src/main/java/com/webonise/tomcat8/redisession/RedisPoolFactory.java

```diff
@@ -0,0 +1,94 @@
+package com.webonise.tomcat8.redisession;
+
+import java.util.Enumeration;
+import java.util.Hashtable;
+import java.util.Map;
+import java.util.Objects;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.naming.Context;
+import javax.naming.Name;
+import javax.naming.NamingException;
+import javax.naming.RefAddr;
+import javax.naming.Reference;
+import javax.naming.spi.ObjectFactory;
+
+import com.webonise.tomcat8.redisession.redisclient.Redis;
+
+import redis.clients.jedis.Protocol;
+
+public class RedisPoolFactory implements ObjectFactory {
+
+	  public static final String DEFAULT_KEY_HOST = "host";
+	  public static final String DEFAULT_KEY_PORT = "port";
+	  public static final String DEFAULT_KEY_TIMEOUT =  "timeout";
+	  private static Logger logger = Logger.getLogger(RedisPoolFactory.class.getName());
```

Comment:
This should be `private static final`.  Also, use Apache Commons Logging, because that's what Tomcat uses: https://tomcat.apache.org/tomcat-8.0-doc/logging.html